zerver/actions/create_user.py

```diff
@@ -349,9 +350,13 @@ def process_new_human_user(
         flags=F("flags").bitor(UserMessage.flags.starred)
     )
 
-    # The 'visibility_policy_banner' is only displayed to existing users.
-    # Mark it as read for a new user.
-    OnboardingStep.objects.create(user=user_profile, onboarding_step="visibility_policy_banner")
+    # If the user selected to import settings from existing account,
+    # 'copy_onboarding_steps' function already did the needed copying.
+    # Skip this operation in that case.
+    if not settings_imported_from_existing_account:
+        # The 'visibility_policy_banner' is only displayed to existing users.
+        # Mark it as read for a new user.
+        OnboardingStep.objects.create(user=user_profile, onboarding_step="visibility_policy_banner")
```

Comment:
I think it might be better to just either catch `IntegrityError` or call `get_or_create`; that makes the logic not sensitive to how the thing might already exist? I think probably for this feature in particular, if they have multiple Cloud accounts, we probably don't want to skip doing this if they haven't seen it in the first realm. (The only effect that might have is making them see it once per org).

Reply:
Made the changes to catch `IntegrityError`